src/data/visualization/VisualizationD2Repository.ts

```diff
@@ -0,0 +1,64 @@
+import { Id } from "../../domain/common/entities/Schemas";
+import { Instance } from "../../domain/instance/entities/Instance";
+import { Visualization } from "../../domain/visualization/entities/Visualization";
+import { VisualizationRepository } from "../../domain/visualization/repositories/VisualizationRepository";
+import { D2Api, MetadataPick } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../../utils/d2-utils";
+
+export class VisualizationD2Repository implements VisualizationRepository {
+    private api: D2Api;
+
+    constructor(instance: Instance) {
+        this.api = getD2APiFromInstance(instance);
+    }
+
+    async getByIds(ids: Id[]): Promise<Visualization[]> {
+        const d2Visualizations = await this.api.models.visualizations
+            .get({
+                fields: visualizationFields,
+                filter: { id: { in: ids } },
+            })
+            .getData();
+
+        return d2Visualizations.objects.map(d2Visualization => this.mapD2VisualizationToVisualization(d2Visualization));
+    }
+
+    private mapD2VisualizationToVisualization(d2Visualization: D2Visualization): Visualization {
+        return {
+            id: d2Visualization.id,
+            rows: (d2Visualization.rows as D2VisualizationRow[]).map(row => ({
```

Comment:
We need a comment here about needing this type assertion. I see that we have `rows: unknown[];  series: unknown[];`. This is something that will need fixing in d2-api generate-schemas.